airbyte-webapp/src/hooks/services/ConnectionEdit/ConnectionEditService.tsx

```diff
@@ -0,0 +1,69 @@
+import { useContext, useState, createContext, useCallback } from "react";
+import { useAsyncFn } from "react-use";
+
+import { ConnectionStatus, WebBackendConnectionUpdate } from "core/request/AirbyteClient";
+
+import { ConnectionFormServiceProvider } from "../ConnectionForm/ConnectionFormService";
+import { useGetConnection, useUpdateConnection, useWebConnectionService } from "../useConnectionHook";
+
+interface ConnectionEditProps {
+  connectionId: string;
+}
+
+const useConnectionEdit = ({ connectionId }: ConnectionEditProps) => {
+  const [connection, setConnection] = useState(useGetConnection(connectionId));
+  const connectionService = useWebConnectionService();
+  const [schemaHasBeenRefreshed, setSchemaHasBeenRefreshed] = useState(false);
+
+  // TODO: Pull error out here, utilize down-line. Maybe use schema error component like on the create page?
+  const [{ loading: schemaRefreshing }, refreshSchema] = useAsyncFn(async () => {
+    const refreshedConnection = await connectionService.getConnection(connectionId, true);
+    setConnection(refreshedConnection);
+    setSchemaHasBeenRefreshed(true);
+  }, [connectionId]);
+
+  const { mutateAsync: updateConnectionAction, isLoading: connectionUpdating } = useUpdateConnection();
+
+  const updateConnection = useCallback(
+    async (connection: WebBackendConnectionUpdate) => {
+      // TODO: Check if the form is dirty before firing off an update action
```

Comment:
Unsure. This only becomes an issue if you change things with a changed stream. Otherwise the form is fine.
It merit's a further conversation I think.

Reply:
This was fixed by removing the `name` from initial values for replication, so it's no longer changed.